include/fixed_containers/integer_range.hpp

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <cassert>
+#include <concepts>
+#include <cstddef>
+
+namespace fixed_containers
+{
+template <typename T>
+concept IsIntegerRange = requires(const T t, const T t2, std::size_t i) {
+    {
+        t.start_inclusive()
+    } -> std::same_as<std::size_t>;
+
+    {
+        t.end_exclusive()
+    } -> std::same_as<std::size_t>;
+
+    {
+        t.contains(i)
+    } -> std::same_as<bool>;
+
+    {
+        t == t2
+    } -> std::same_as<bool>;
+};
+
+template <std::size_t START_INCLUSIVE, std::size_t END_EXCLUSIVE>
+    requires(START_INCLUSIVE <= END_EXCLUSIVE)
+class CompileTimeIntegerRange
```

Comment:
Why can't we just use one class for integer range that supports a `constexpr` constructor? Seems weird to have two implementations, where only one is "compile-time."

Reply:
Takes up less space, so the iterator is a bit more lightweight:
```C++
static_assert(sizeof(IntegerRange) == 16);
static_assert(sizeof(CompileTimeIntegerRange<0, 3>) == 1);
```

Additionally, since it is "consteval", the compiler *might* be able to apply more optimizations, like for example removing the `i >= starting_index` check if `starting_index` is 0, because it is guaranteed to always be true (`std::size_t`).

Maybe `CompileTime` is not expressive enough for this.